pkg/apiserver/apiserver.go

```diff
@@ -131,6 +134,7 @@ func (as *APIServer) Stop() {
 }
 
 func (as *APIServer) isReady() (bool, error) {
+	apiServiceName := getEnvVal(kappctrlAPIEnvKey, defaultApiServiceName)
```

Comment:
@linkdd whast the thinking behind `v1alpha1.data.packaging.carvel.dev` override? this is a name that is used by packages.

Reply:
I'm currently working on an Helm Chart (see #283). This would allow me to have multiple release of the same chart in the same cluster (not that's something you would want, but who knows with multi-tenancy?).

By overriding this value, one could provide a `v1alpha1` and a `v1alpha2` without conflicts in the chart. Useful for A/B testing?

The `APIService` is required by the operator to work. So if only one chart in the cluster can deploy it (because of the static name), this introduces a tight coupling between the operator and how it is deployed.

Reply:
> By overriding this value, one could provide a v1alpha1 and a v1alpha2 without conflicts in the chart. Useful for A/B testing?

in future versions of kapp-controller i would expect kc to support multiple versions at the same time for backwards compatible upgrades (standard behaviour for k8s controllers; resources within the cluster and in different packages would be using different versions in parallel). so i dont think this really works by making this configurable.

Reply:
Ok, I removed it and will add a condition in the Helm Chart to skip the creation of the APIService.

This means that if there are multiple release of the same chart in the cluster, the `Package` resources will be served by only one of them (is this an issue?)

Reply:
> This means that if there are multiple release of the same chart in the cluster, the Package resources will be served by only one of them (is this an issue?)

i dont think we really can support multiple kapp-controllers on one cluster without dynamic renaming of CRDs and APIServices (and that's very uncommon in k8s ecosystem). renaming leads to a ton of complexity/confusion since rename has to propagate not thru all the code but also into consumer configuration (e.g. wouldnt be able to use regular App CR api group, or package repositories wouldnt be able to use packaging apiGroups). if you want to reliably do blue-green testing for kapp-controller, you ll have to do it against multiple clusters (eg staging vs prod).

Reply:
Fair enough, I'll just have to document my chart and make it clear that it cannot be deployed multiple times.

So now, this PR only introduce the dynamic service name. Hope it can be merged 🙂

Reply:
@linkdd to push further on this... sorry to be bombarding you with questions... how are you planning to take advantage of changeable service name? any specific use case for allowing it to be renamed?

Reply:
@cppforlife At the moment, in my [Helm Chart](https://github.com/link-society/charts/blob/incubator/kapp-controller/incubator/kapp-controller/templates/service.yml), the `Service` is named `{{ include "kapp-controller.fullname" . }}-packaging-api`.
 
 Since the controller can be deployed in any namespace this avoid name clashes (what if I already have a service named `packaging-api` from a completely different project?).

The ytt templates used to generate the release manifest defines the namespace as `kapp-controller` by default, but it can be overridden. For Helm, the default namespace is the one specified in your `.kube/config`, so one should be careful about that 🙂